src/recovery/mod.rs

```diff
@@ -354,7 +356,27 @@ impl Recovery {
             handshake_completed,
         );
 
-        self.loss_probes[epoch] = 2;
+        let unacked_iter = self.sent[epoch]
+            .iter_mut()
+            // Skip packets that have already been acked or lost, and packets
+            // that don't contain either CRYPTO or STREAM frames.
+            .filter(|p| p.has_data && p.time_acked.is_none() && p.time_lost.is_none())
```

Comment:
It has been same from previous commit, but I am just thinking of `p.time_lost.is_none()` condition is necessary or not. This will skip packets declared as lost for a Probe packet.

However when a Probe packet is scheduled, I expect there is already some data frame marked as lost (by nature) and those unacked and lost data frames may need to be retransmitted first, since those are usually older than unacked & non-lost ones. Also when 2nd PTO timeout all send buffer is already lost, then we may end up sending PING instead of retransmitting oldest unacked one.

Reply:
This is interesting topic for me but not a main topic of this PR, so let me dig into a little bit and create a new PR if needed.

Reply:
One thing to note is that when a packet is marked as lost, its frames are moved out of the `Sent` object (as opposed to just being cloned as in the PTO case), so we can't actually schedule them again here.

Though I guess the main point is whether we should re-schedule lost data again here. If the packet is known as lost than that data is already re-scheduled, are you saying we should re-schedule it again?

Reply:
> If the packet is known as lost than that data is already re-scheduled, are you saying we should re-schedule it again?
Assuming we lost all the ack and let's say we send data 1, after some time we mark data 1 as lost, retransmit, still getting no ack, and then PTO is fired. My point is that in this case this lost data 1 need to be retransmit in the PTO, if rescheduled already but not sent yet.